serde/src/test/kotlin/com/amazon/ionhiveserde/Util.kt

```diff
@@ -15,7 +15,53 @@
 
 package com.amazon.ionhiveserde
 
+import com.amazon.ion.IonDatagram
+import com.amazon.ion.IonSequence
+import com.amazon.ion.IonStruct
 import com.amazon.ion.system.IonSystemBuilder
+import com.amazon.ionhiveserde.caseinsensitivedecorator.IonSequenceCaseInsensitiveDecorator
+import com.amazon.ionhiveserde.caseinsensitivedecorator.IonStructCaseInsensitiveDecorator
 
 internal val ION = IonSystemBuilder.standard().build()
 internal val ionNull = ION.newNull()
+
+internal fun datagram_for(s: String): IonDatagram {
+    return ION.loader.load(s)
+}
+
+internal fun struct_for(s: String): IonStruct {
+    val v = datagram_for(s).iterator().next()
+    if (v !is IonStruct) throw IllegalArgumentException("Required an IonStruct, found ${v.javaClass.simpleName}")
+
+    return v
+}
+
+internal fun case_insensitive_decorator_struct_for(s : String): IonStruct {
+    return IonStructCaseInsensitiveDecorator(struct_for(s))
+}
+
+internal fun sequence_for(s: String): IonSequence {
+    val v = datagram_for(s).iterator().next()
+    if (v !is IonSequence) throw IllegalArgumentException("Required an IonSequence, found ${v.javaClass.simpleName}")
+
+    return v
+}
+
+internal fun case_insensitive_decorator_sequence_for(s : String): IonSequence {
+    return IonSequenceCaseInsensitiveDecorator(sequence_for(s))
+}
```

Comment:
This could be replaced with the following, then inlined:

```kotlin
IonCaseInsensitiveDecorator.valueFor(sequence_for(s));
```

If that's too verbose to read in our tests then you can do this:

```kotlin
internal fun case_insensitive(v: IonValue) {
  IonCaseInsensitiveDecorator.wrapValue(v)
}

// use like this:
case_insensitive(struct_for("{ foo: 1, bar: 2 }"));
```

This reads nearly the same, but uses a more general mechanism.

Reply:
Addressed with the second option. 

`case_insensitive` is more generic and works for both struct and sequence.